pvlib/iotools/pvgis.py

```diff
@@ -390,9 +391,32 @@ def read_pvgis_hourly(filename, pvgis_format=None, map_variables=True):
     raise ValueError(err_msg)
 
 
+def _coerce_and_rotate_pvgis(pvgis_data, tz, year):
+    """
+    After converting TZ, rotate indices so timeseries starts at midnight
+    and force all indices to a common year. Only works for integer TZ.
+    """
+    # assert tz//1 == tz
+    tzname = f'Etc/GMT{-tz:+d}'
+    # check if February is leap year
+    feb1 = pvgis_data[pvgis_data.index.month == 2].index.year[0]
+    if calendar.isleap(feb1):
+        # replace Feb year with a non-leap year
+        pvgis_data.index = [
+            ts.replace(year=1990) if ts.month == 2 else ts
+            for ts in pvgis_data.index]
+    pvgis_data = pd.concat(
+        [pvgis_data.iloc[-tz:], pvgis_data.iloc[:-tz]],
+        axis=0).tz_convert(tzname)
```

Comment:
I kind of liked the `np.roll()` solution but in python there are always _so_ many ways to achieve the same result!

Reply:
I don’t recall why I chose this approach but I can test if `np.roll` works better

Reply:
it works!